wordpress-mobile/WordPress-iOS
# Install Danger locally for better performances on CI

https://github.com/wordpress-mobile/WordPress-iOS/pull/13423 added two [GitHub workflows](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions) to run static checks on PRs via [Danger](https://danger.systems/js/).

To keep the setup simple, I decided to use the [Danger GitHub Action](https://github.com/marketplace/actions/danger-js-action), but the time it takes to run on CI is too much, more than a minute.

This diff switches the setup to a local installation of Danger, which runs on CI in less than 30 seconds –after the first run, when the cache has been build.

Before | After | After, with cache
--- | --- | ---
<img width="647" alt="Screen Shot 2020-02-17 at 11 17 21" src="https://user-images.githubusercontent.com/1218433/74615716-42afd880-5177-11ea-8ccd-a849820769e4.png"> | <img width="631" alt="Screen Shot 2020-02-17 at 11 15 26" src="https://user-images.githubusercontent.com/1218433/74615718-45123280-5177-11ea-9f38-6f46c36161cf.png"> | <img width="647" alt="Screen Shot 2020-02-17 at 11 18 37" src="https://user-images.githubusercontent.com/1218433/74615720-47748c80-5177-11ea-9f98-633a4149f688.png">


This approach adds a bit of overhead on the repo: another dependency manager. Devs don't need to interact with it though, only CI does. I think it's an acceptable tradeoff to deliver feedback on the PR faster.

### How to test

Because this Danger setup runs on [pull request events](https://help.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-event-pull_request), you're better off testing this changes in a new PR. It will keep this PR cleaner. Feel free to use https://github.com/mokagio/WordPress-iOS/pull/9 as a base for it, the code is the same, just less tidy.

The test strategy should be the same as #13442:

- Add and remove labels to verify Danger leaves comments when the PR has no label.
- Change the codebase in a way that will trigger a warning or failure in the consistency checks.
